### ISSUE TITLE
Syslog logger not working?

### ISSUE THREAD

**Issue by juju4** [facebook/osquery#3199]
I'm testing osqueryd with the different logger, filesystem, tls and syslog

I found that syslog output does not seem to work.
Install is done with ansible https://github.com/juju4/ansible-osquery/tree/devel
resulting rsyslog config from template
```
19.*      /var/log/osquery_syslog-results.log
& stop
if $programname == 'osqueryd' then           /var/log/osquery_syslog-prog.log
& stop
```
https://travis-ci.org/juju4/ansible-osquery/jobs/224808301#L4186
https://travis-ci.org/juju4/ansible-osquery/jobs/224817817#L4231

In some of my tests (vagrant/virtualbox and no docker systemd troubles), I got daemon info into /var/log/osquery_syslog-prog.log but nothing in results file (with rsyslogd running).
Same result with remote udp syslog.

Any options to troubleshoot?

**Comment by theopolis** [comment]
I haven't looked too deeply, but want to make sure `rsyslog` is configured to receive `LOG_INFO`.

From: http://osquery.readthedocs.io/en/stable/installation/cli-flags/#loggingresults-flags

> --logger_syslog_facility
> 
> Set the syslog facility (number) 0-23 for the results log. When using the syslog logger plugin the default facility is 19 at the LOG_INFO level, which does not log to /var/log/system.
> 
> --logger_syslog_prepend_cee
> 
> Prepend a [at]cee: cookie to JSON-formatted messages sent to the syslog logger plugin. Several syslog parsers use this cookie to indicate that the message payload is parseable JSON. The default value is false.

**Comment by juju4** [comment]
On rsyslog side, that was this worry that made me add an explicit rule to a defined target (remote server or distinct file) for "19.*" vs "*.*". just tried with "local3.*" just in case.

I don't see in default rsyslog conf (trusty or xenial) any rules to drop local3 logs, just to prevent them to be in main log file (syslog or messages depending on platform)

**Comment by juju4** [comment]
I think I locate the problem to 2 issues
- osquery.conf overrides logger passed in osquery.flags. reading the docs (https://osquery.readthedocs.io/en/stable/deployment/logging/) doesn't give this view clearly.
- syslog logging works if alone in osquery.conf. passing "filesystem,syslog" in osquery.conf results in "filesystem" only...

**Comment by theopolis** [comment]
I think this should be fixed in #3488, let me know otherwise :)
